### DISCUSSION TITLE
An unexpected error about the boundary setting in the mesh

### DISCUSSION THREAD

**Discussion by watercloud0** [idaholab/moose#23778] [Q&A Meshing] [answered]
Hi, I would like to ask a question about the mesh boundary conditions.
When running the input file, there is an error as shown in the figure (the following nodes sets do not exist on the mesh:temp_fixed),
<img width="366" alt="0fd2783f758a21b091419fc2aecc578" src="https://user-images.githubusercontent.com/96585522/226186117-e38bcbc6-0898-4e76-b58f-a80eb35c1ee2.png">

However, I checked in paraview that the temp_fixed set for the corresponding side_sets exists (as shown in the figure below).
<img width="178" alt="a1a5973e001774bac2701735d7df25d" src="https://user-images.githubusercontent.com/96585522/226186135-a999179d-6785-4a44-90dd-64c8b160f64c.png">

This problem also exists for the pin_fiexed boundary.
 What could be the reason for this? How to solve this problem?
The locations of temp_fixed, pin_fixed correspond to the following two figures respectively.
<img width="261" alt="5f8a7fd7c629c1d639606aaa5644943" src="https://user-images.githubusercontent.com/96585522/226186141-d5467302-533f-4110-b85f-2e1cffa2a859.png">
<img width="312" alt="b141ee73744d320a77e508c69da416f" src="https://user-images.githubusercontent.com/96585522/226186148-a92c3256-2c0a-455b-a3b7-3e208d425719.png">

The following figure shows the settings of the mesh in the input file.
<img width="482" alt="faa74d90c9f57c83dd8e91f47c42704" src="https://user-images.githubusercontent.com/96585522/226186153-976d1c7a-2090-4ccb-9414-538636bb3b08.png">

**Comment by GiudGiud** [comment, marked as answer]
Hello

there s a conversion parameter in the Mesh block you can add to convert side sets to node sets

something like convert_sidelist_to_nodelist. You ll want to check the documentation for the exact spelling


Guillaume

**Reply by watercloud0** [reply]
Thank you for your answer, I will look up this knowledge.

**Reply by GiudGiud** [reply]
construct_node_list_from_side_list is the name of the parameter, and it should default to True

But actually I think what may be failing here is the file reading.
Can you try using this instead
https://mooseframework.inl.gov/source/meshgenerators/FileMeshGenerator.html

**Reply by watercloud0** [reply]
Ok I will try this and thank you very much！

**Reply by watercloud0** [reply]
Hi, I still can't solve this problem, maybe I should provide more complete information, here is the complete run through, and the boundary condition settings on line 48.It seems to finish reading the grid.
<img width="513" alt="dfbf017a780eb634565fd1fd269ffa8" src="https://user-images.githubusercontent.com/96585522/226413793-5a6f97f3-16b8-466c-8134-331a6a12a45b.png">
<img width="225" alt="861989a406133127e138aaf20041820" src="https://user-images.githubusercontent.com/96585522/226414040-38896122-5dc7-46bc-985c-f4d2ffd1310d.png">

Can I change the corresponding reported error from side_sets（temp_fixed、pin_fixed） to node_sets by modifying the boundary settings of the grid?

**Reply by GiudGiud** [reply]
what is your Mesh block
can you add show_info = true to the mesh generator and look at the console output

**Reply by watercloud0** [reply]
The grid file information for paraview is displayed as follows.
<img width="247" alt="427c756ae8f0d52b60685a2dbfe2fed" src="https://user-images.githubusercontent.com/96585522/226422738-6f9ff64d-2ca4-4f8b-ba52-7fb0d9951276.png">
<img width="196" alt="ae36c8a88bab6ec9097f17dcea75f10" src="https://user-images.githubusercontent.com/96585522/226423308-24112227-fb38-4fc1-a7a7-48a8b943d86b.png">

In addition, I found that if I don't define the corresponding bounds in [BCS], no error is reported, but the result is not produced for a long time and the process is killed.
I'm very sorry, but it's late at night where I am and I'm going to bed, so thank you for your help today!

**Reply by GiudGiud** [reply]
No worries
I m talking about the [Mesh] block in MOOSE, and the mesh generators that are objects in that block

**Reply by watercloud0** [reply]
Thank you very much for the explanation!
This time the error message is different.
 After using FileMeshGenerator the error about the boundary no longer appears and the problem seems to be solved.
However, this time it's the block(pins) error, which represents multiple pins. 
**As shown in the figure.**
<img width="489" alt="f2e367feed42ee6010d1cdd211f3e16" src="https://user-images.githubusercontent.com/96585522/226580128-60837510-91e8-4e5e-9d89-6c34f513c262.png">

**Here is the complete input file code   ：**

     [Mesh]
         [generated_mesh]
            type = FileMeshGenerator
             file = model_HT.e
             show_info = true
         []
     []

    [Variables]
        [T]
          initial_condition = 300
        []
        [disp_x]
        []
        [disp_y]
        []
        [disp_z]
        []
    []

    [Kernels]
        [hc]
            type = HeatConduction
            variable = T
        []
        [TensorMechanics]
            displacements = 'disp_x disp_y disp_z'
        []
        [heat_source]
            type = HeatSource
            variable = T
            block = chip
            value = 400
        []
    []

    [BCs]  
        [temp_heat_flux]
            type = ConvectiveHeatFluxBC
            variable = T
            boundary = heat_flux
            T_infinity = 303.15
            heat_transfer_coefficient = 50
        []
        [temp_temp_fixed]
            type = DirichletBC
            variable = T
            boundary = temp_fixed
            value = 323.15
        []
        [pin_fixed_disp_y]
            type = DirichletBC
            variable = disp_y
            boundary = pin_fixed
            value = 0
        []
        [pin_fixed_disp_x]
            type = DirichletBC
            variable = disp_x
            boundary = pin_fixed
            value = 0
        [] 
        [pin_fixed_disp_z]
            type = DirichletBC
            variable = disp_z
            boundary = pin_fixed
            value = 0
        []                
    []

    [Materials]
            [cond_package]
                type = GenericConstantMaterial                            
                block = package  
                prop_names = thermal_conductivity                       
                prop_values = 0.2 
            []
            [cond_pins]
                type = GenericConstantMaterial                            
                block = pins  
                prop_names = thermal_conductivity                       
                prop_values = 238 
            []  
            [cond_wire]
                type = GenericConstantMaterial                            
                block = wire  
                prop_names = thermal_conductivity                       
                prop_values = 400 
            []
            [cond_chip]
                type = GenericConstantMaterial                            
                block = chip  
                prop_names = thermal_conductivity                       
                prop_values = 130 
            []  
            [cond_pcb]
                type = GenericConstantMaterial                            
                block = pcb  
                prop_names = thermal_conductivity                       
                prop_values = 0.3 
            []  
            [cond_regulator]
                type = GenericConstantMaterial                            
                block = regulator_block  
                prop_names = thermal_conductivity                       
                prop_values = 400 
            []
            
            [cond_temp_boundary]
                type = GenericConstantMaterial                            
                block = temp_boundary  
                prop_names = thermal_conductivity                       
                prop_values = 400 
            []

        

            [elasticity_tensor_package]
                type = ComputeIsotropicElasticityTensor
                youngs_modulus = 10e9
                poissons_ratio = 0.38
                block = package
            []
            [elasticity_tensor_pins]
                type = ComputeIsotropicElasticityTensor
                youngs_modulus = 70e9
                poissons_ratio = 0.33
                block = pins
            []
            [elasticity_tensor_wire]
                type = ComputeIsotropicElasticityTensor
                youngs_modulus = 110e9
                poissons_ratio = 0.35
                block = wire
            []
            [elasticity_tensor_chip]
                type = ComputeIsotropicElasticityTensor
                youngs_modulus = 170e9
                poissons_ratio = 0.28
                block = chip
            []
            [elasticity_tensor_pcb]
                type = ComputeIsotropicElasticityTensor
                youngs_modulus = 22e9
                poissons_ratio = 0.15
                block = pcb
            []
            [elasticity_tensor_regulator]
                type = ComputeIsotropicElasticityTensor
                youngs_modulus = 110e9
                poissons_ratio = 0.35
                block = regulator_block
            [] 

            #!!!!
            [elasticity_tensor_temp_boundary]
                type = ComputeIsotropicElasticityTensor
                youngs_modulus = 110e9
                poissons_ratio = 0.35
                block = temp_boundary
            []               

            [thermal_strain_package]
                type = ComputeThermalExpansionEigenstrain
                thermal_expansion_coeff = 56.8e-6
                temperature = T
                stress_free_temperature = 300
                eigenstrain_name = eigenstrain_package
                block = package
            []
            [thermal_strain_pins]
                type = ComputeThermalExpansionEigenstrain
                thermal_expansion_coeff = 23e-6
                temperature = T
                stress_free_temperature = 300
                eigenstrain_name = eigenstrain_pins
                block = pins
            []
            [thermal_strain_wire]
                type = ComputeThermalExpansionEigenstrain
                thermal_expansion_coeff = 17e-6
                temperature = T
                stress_free_temperature = 300
                eigenstrain_name = eigenstrain_wire
                block = wire
            []
            [thermal_strain_chip]
                type = ComputeThermalExpansionEigenstrain
                thermal_expansion_coeff = 2.6e-6
                temperature = T
                stress_free_temperature = 300
                eigenstrain_name = eigenstrain_chip
                block = chip
            []
            [thermal_strain_pcb]
                type = ComputeThermalExpansionEigenstrain
                thermal_expansion_coeff = 18e-6
                temperature = T
                stress_free_temperature = 300
                eigenstrain_name = eigenstrain_pcb
                block = pcb
            []
            [thermal_strain_regulator]
                type = ComputeThermalExpansionEigenstrain
                thermal_expansion_coeff = 17e-6
                temperature = T
                stress_free_temperature = 300
                eigenstrain_name = eigenstrain_regulator
                block = regulator_block
            []
            [thermal_strain_temp_boundary]
                type = ComputeThermalExpansionEigenstrain
                thermal_expansion_coeff = 17e-6
                temperature = T
                stress_free_temperature = 300
                eigenstrain_name = eigenstrain_temp_boundary
                block = temp_boundary
            []

            [strain_package]
                type = ComputeSmallStrain
                displacements = 'disp_x disp_y disp_z'
                eigenstrain_names = 'eigenstrain_package'
                block = package
            []
            [strain_pins]
                type = ComputeSmallStrain
                displacements = 'disp_x disp_y disp_z'
                eigenstrain_names = 'eigenstrain_pins'
                block = pins
            []
            [strain_wire]
                type = ComputeSmallStrain
                displacements = 'disp_x disp_y disp_z'
                eigenstrain_names = 'eigenstrain_wire'
                block = wire
            []
            [strain_chip]
                type = ComputeSmallStrain
                displacements = 'disp_x disp_y disp_z'
                eigenstrain_names = 'eigenstrain_chip'
                block = chip
            []
            [strain_regulator]
                type = ComputeSmallStrain
                displacements = 'disp_x disp_y disp_z'
                eigenstrain_names = 'eigenstrain_regulator'
                block = regulator_block
            []

            #!!!!
            [strain_temp_boundary]
                type = ComputeSmallStrain
                displacements = 'disp_x disp_y disp_z'
                eigenstrain_names = 'eigenstrain_temp_boundary'
                block = temp_boundary
            []

        [stress] #We use linear elasticity
            type = ComputeLinearElasticStress
        []
    []

    [Executioner]
      type = Steady
      #Preconditioned JFNK (default)
      solve_type = 'PJFNK'
      petsc_options_iname = '-pc_type -pc_hypre_type -ksp_gmres_restart'
      petsc_options_value = 'hypre boomeramg 101'
    
      l_max_its = 30 
      nl_max_its = 100 
      nl_abs_tol = 1e-9 
      l_tol = 1e-04 
    []  

    [Preconditioning]
      [SMP]
        type = SMP
        full = true
      []
    []

      [Outputs]
        exodus = true
      []

**The console outputs the following message.**
*** Info ***
TensorMechanics Action: selecting 'total small strain' formulation. Use `incremental = true` to select 'incremental small strain' instead.
Setting Up.........
FileMeshGenerator 'generated_mesh':  Mesh Information:
FileMeshGenerator 'generated_mesh':   elem_dimensions()={2, 3}
FileMeshGenerator 'generated_mesh':   spatial_dimension()=3
FileMeshGenerator 'generated_mesh':   n_nodes()=2173176
FileMeshGenerator 'generated_mesh':     n_local_nodes()=2173176
FileMeshGenerator 'generated_mesh':   n_elem()=2060989
FileMeshGenerator 'generated_mesh':     n_local_elem()=2060989
FileMeshGenerator 'generated_mesh':     n_active_elem()=2060989
FileMeshGenerator 'generated_mesh':   n_subdomains()=7
FileMeshGenerator 'generated_mesh':   n_elemsets()=0
FileMeshGenerator 'generated_mesh':   n_partitions()=1
FileMeshGenerator 'generated_mesh':   n_processors()=1
FileMeshGenerator 'generated_mesh':   n_threads()=1
FileMeshGenerator 'generated_mesh':   processor_id()=0
FileMeshGenerator 'generated_mesh':   is_prepared()=true
FileMeshGenerator 'generated_mesh':   is_replicated()=true
FileMeshGenerator 'generated_mesh': 
FileMeshGenerator 'generated_mesh':  Mesh Bounding Box:
FileMeshGenerator 'generated_mesh':   Minimum: (x,y,z)=(     -10,       -5,     -1.9)
FileMeshGenerator 'generated_mesh':   Maximum: (x,y,z)=(      10,        5,     0.75)
FileMeshGenerator 'generated_mesh':   Delta:   (x,y,z)=(      20,       10,     2.65)
FileMeshGenerator 'generated_mesh': 
FileMeshGenerator 'generated_mesh':  Mesh Element Type(s):
FileMeshGenerator 'generated_mesh':   HEX8, QUADSHELL4
FileMeshGenerator 'generated_mesh': 
FileMeshGenerator 'generated_mesh':  Mesh Nodesets:
FileMeshGenerator 'generated_mesh':   None
FileMeshGenerator 'generated_mesh': 
FileMeshGenerator 'generated_mesh':  Mesh Sidesets:
FileMeshGenerator 'generated_mesh':   Sideset 1 (Pin_fixed), 24 sides (QUAD4), 24 elems (HEX8), 36 nodes
FileMeshGenerator 'generated_mesh':    Side volume: 0.064
FileMeshGenerator 'generated_mesh':    Bounding box minimum: (x,y,z)=(  -4.645,    -2.85,     -0.9)
FileMeshGenerator 'generated_mesh':    Bounding box maximum: (x,y,z)=(  -4.245,    -2.69,     -0.9)
FileMeshGenerator 'generated_mesh':    Bounding box delta: (x,y,z)=(     0.4,     0.16, 4.44089e-16)
FileMeshGenerator 'generated_mesh':   Sideset 3 (heat_flux), 5304 sides (QUAD4), 5304 elems (HEX8), 5832 nodes
FileMeshGenerator 'generated_mesh':    Side volume: 13.318
FileMeshGenerator 'generated_mesh':    Bounding box minimum: (x,y,z)=(  -4.645,       -5,     -1.9)
FileMeshGenerator 'generated_mesh':    Bounding box maximum: (x,y,z)=(      10,        5,     0.01)
FileMeshGenerator 'generated_mesh':    Bounding box delta: (x,y,z)=(  14.645,       10,     1.91)
FileMeshGenerator 'generated_mesh': 
FileMeshGenerator 'generated_mesh':  Mesh Edgesets:
FileMeshGenerator 'generated_mesh':   None
FileMeshGenerator 'generated_mesh': 
FileMeshGenerator 'generated_mesh':  Mesh Subdomains:
FileMeshGenerator 'generated_mesh':   Subdomain 1 (package): 455520 elems (HEX8, 455520 active), 478632 active nodes
FileMeshGenerator 'generated_mesh':    Volume: 54.9549
FileMeshGenerator 'generated_mesh':    Bounding box minimum: (x,y,z)=(   -4.95,    -1.95,    -0.75)
FileMeshGenerator 'generated_mesh':    Bounding box maximum: (x,y,z)=(    4.95,     1.95,     0.75)
FileMeshGenerator 'generated_mesh':    Bounding box delta: (x,y,z)=(     9.9,      3.9,      1.5)
FileMeshGenerator 'generated_mesh':   Subdomain 2 (Pins): 14560 elems (HEX8, 14560 active), 21141 active nodes
FileMeshGenerator 'generated_mesh':    Volume: 1.90478
FileMeshGenerator 'generated_mesh':    Bounding box minimum: (x,y,z)=(  -4.645,    -2.85,     -0.9)
FileMeshGenerator 'generated_mesh':    Bounding box maximum: (x,y,z)=(   4.645,     2.85,      0.1)
FileMeshGenerator 'generated_mesh':    Bounding box delta: (x,y,z)=(    9.29,      5.7,        1)
FileMeshGenerator 'generated_mesh':   Subdomain 3 (wire): 944 elems (HEX8, 944 active), 2142 active nodes
FileMeshGenerator 'generated_mesh':    Volume: 0.02358
FileMeshGenerator 'generated_mesh':    Bounding box minimum: (x,y,z)=(  -4.645,    -1.95,        0)
FileMeshGenerator 'generated_mesh':    Bounding box maximum: (x,y,z)=(    -0.5,      0.2,     0.01)
FileMeshGenerator 'generated_mesh':    Bounding box delta: (x,y,z)=(   4.145,     2.15,     0.01)
FileMeshGenerator 'generated_mesh':   Subdomain 4 (chip): 1160 elems (HEX8, 1160 active), 1701 active nodes
FileMeshGenerator 'generated_mesh':    Volume: 0.1
FileMeshGenerator 'generated_mesh':    Bounding box minimum: (x,y,z)=(    -0.5,     -0.5,    -0.05)
FileMeshGenerator 'generated_mesh':    Bounding box maximum: (x,y,z)=(     0.5,      0.5,     0.05)
FileMeshGenerator 'generated_mesh':    Bounding box delta: (x,y,z)=(       1,        1,      0.1)
FileMeshGenerator 'generated_mesh':   Subdomain 5 (PCB): 1579280 elems (HEX8, 1579280 active), 1670865 active nodes
FileMeshGenerator 'generated_mesh':    Volume: 200
FileMeshGenerator 'generated_mesh':    Bounding box minimum: (x,y,z)=(     -10,       -5,     -1.9)
FileMeshGenerator 'generated_mesh':    Bounding box maximum: (x,y,z)=(      10,        5,     -0.9)
FileMeshGenerator 'generated_mesh':    Bounding box delta: (x,y,z)=(      20,       10,        1)
FileMeshGenerator 'generated_mesh':   Subdomain 6 (Regulator): 6325 elems (QUADSHELL4, 6325 active), 6496 active nodes
FileMeshGenerator 'generated_mesh':    Volume: 15.936
FileMeshGenerator 'generated_mesh':    Bounding box minimum: (x,y,z)=(      -8,       -5,     -0.9)
FileMeshGenerator 'generated_mesh':    Bounding box maximum: (x,y,z)=(      -4,       -1,     -0.9)
FileMeshGenerator 'generated_mesh':    Bounding box delta: (x,y,z)=(       4,        4, 4.44089e-16)
FileMeshGenerator 'generated_mesh':   Subdomain 7 (Temp_Boundary): 3200 elems (QUADSHELL4, 3200 active), 3321 active nodes
FileMeshGenerator 'generated_mesh':    Volume: 8
FileMeshGenerator 'generated_mesh':    Bounding box minimum: (x,y,z)=(     -10,       -5,     -0.9)
FileMeshGenerator 'generated_mesh':    Bounding box maximum: (x,y,z)=(      -8,       -1,     -0.9)
FileMeshGenerator 'generated_mesh':    Bounding box delta: (x,y,z)=(       2,        4, 4.44089e-16)
FileMeshGenerator 'generated_mesh':   Global mesh volume = 280.919

  Setting Up Undisplaced Mesh
    Preparing Mesh
      Updating Mesh                                                                      [  8.92 s] [   44 MB]
    Finished Preparing Mesh                                                              [  9.38 s] [   44 MB]
  Finished Setting Up Undisplaced Mesh                                                   [  9.38 s] [   44 MB]

**The error is reported as follows.The code object that reports the error is [cond_pins]**

*** ERROR ***
/home/ubuntu2023/projects/moose/modules/combined/Mesh.i:80: (Materials/cond_pins/block):
    the following blocks (ids) do not exist on the mesh: pins (65535)

application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
[unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=1
:
system msg for write_line failure : Bad file descriptor

**Reply by GiudGiud** [reply]
You spelled it pins in an object when the block name is actually Pins

**Reply by watercloud0** [reply]
Yes, I noticed the problem and corrected it, it seems that the names in the paraview and the info in the console don't match. The previous Pin_fixed boundary error should be the same problem.
But why the information about Sideset 2-temp_fixed is not shown in Mesh Sidesets?
The error reported before appears again.
<img width="425" alt="602c791074d596a769abcefa552660f" src="https://user-images.githubusercontent.com/96585522/226672051-52a1e773-ac49-4206-80a8-e7f920fcb89d.png">

**Reply by GiudGiud** [reply]
It doesnt exist, it s not being loaded
```
FileMeshGenerator 'generated_mesh': Mesh Nodesets:
FileMeshGenerator 'generated_mesh': None
FileMeshGenerator 'generated_mesh':
```

you can look at recreating it using the moose mesh generators if you re tired of figuring out why it s not being loaded.
https://mooseframework.inl.gov/syntax/Mesh/index.html

**Reply by watercloud0** [reply]
Ok, thanks for your help, I will check this link.
